lib/generators/trk_datatables/trk_datatables_generator.rb

```diff
@@ -1,58 +1,72 @@
-module TrkDatatables
-  module Generators
-    class TrkDatatablesGenerator < Rails::Generators::NamedBase
-      # we can call with `rails g trk_datatables` instead of: `rails g trk_datatables:trk_datatables`
-      namespace "trk_datatables"
-      source_root File.expand_path("../templates", __dir__)
-
-      desc "Generates datatables file for a give NAME"
-      def create
-        begin
-          class_name.constantize
-          @trk_class_name = "#{class_name.pluralize}Datatable"
-          @trk_file_name = "#{plural_name}_datatable"
-        rescue NameError => e
-          Rails.logger.info e.message
-          @skip_model = true
-          @trk_class_name = "#{class_name}Datatable"
-          @trk_file_name = "#{singular_name}_datatable"
-        end
+require "rails"
+require "rails/generators"
 
-        template "trk_datatable.rb", "app/datatables/#{@trk_file_name}.rb"
+class TrkDatatablesGenerator < Rails::Generators::NamedBase
+  # we can call with `rails g trk_datatables` instead of: `rails g trk_datatables:trk_datatables`
+  namespace "trk_datatables"
+  source_root File.expand_path("../templates", __dir__)
 
-        say <<~TEXT
-          ======================================================================
-          You can use in your controller
+  class_option :rspec, type: :boolean, required: false
+  class_option :minitest, type: :boolean, required: false
 
-          # app/controllers/#{plural_name}_controller.rb
-          class #{class_name.pluralize}Controller < ApplicationController
-            def index
-              @datatable = #{@trk_class_name}.new view_context
-            end
+  desc "Generates datatables file for a give NAME"
+  def create
+    begin
+      class_name.constantize
+      @trk_class_name = "#{class_name.pluralize}Datatable"
+      @trk_file_name = "#{plural_name}_datatable"
+    rescue NameError => e
+      # Rails.logger.info e.message
+      @skip_model = true
+      @trk_class_name = "#{class_name}Datatable"
+      @trk_file_name = "#{singular_name}_datatable"
+      @test_framework = if options["rspec"].present?
```

Comment:
If there is no option provided, we could determine for example with
```
else
  File.directory?('spec') ? :rspec : :minitest
```

Reply:
I wanted to give user an option to not generate tests by omitting `--rspec` and `--minitest`. But we can add this if you think it's better like that. I was also thinking to set up some config so user can select default test_framework. What do you think is best approach?

Reply:
I wanted to give user an option to not generate tests by omitting `--rspec` and `--minitest`. But we can add this if you think it's better like that. I was also thinking to set up some config so user can select default test_framework. What do you think is best approach?